### PR TITLE
Pomme de terre

### PR DESCRIPTION
**Pull request by fabien-zimmer**

### CODE REVIEW

File: src/main/java/org/taktik/icure/logic/impl/UserLogicImpl.java [hunk 1]
```diff
@@ -754,9 +754,9 @@ public User setProperties(User user, List<Property> properties) {
 			Optional<Property> prop = user.getProperties().stream().filter(pp -> pp.getType().getIdentifier().equals(p.getType().getIdentifier())).findAny();
 
 			if (!prop.isPresent()) {
-				user.getProperties().add(new Property(new PropertyType(p.getType().getType(), p.getType().getIdentifier()), new TypedValue(p.getType().getType(), p.getValue())));
+				user.getProperties().add(new Property(new PropertyType(p.getTypedValue().getType(), p.getType().getIdentifier()), new TypedValue(p.getTypedValue().getType(), p.getValue())));
 			} else {
-				prop.orElseThrow(IllegalStateException::new).setTypedValue(new TypedValue(p.getType().getType(), p.getValue()));
+				prop.orElseThrow(IllegalStateException::new).setTypedValue(new TypedValue(p.getTypedValue().getType(), p.getValue()));
```
Comment (aduchate) [inline src/main/java/org/taktik/icure/logic/impl/UserLogicImpl.java:759]: Why ?

Comment (AssossCare) [reply src/main/java/org/taktik/icure/logic/impl/UserLogicImpl.java:759]: Julien says that when you use boolean value in properties, getType().getType() is null, you must use getTypedValue().getType to have right value.

File: src/main/java/org/taktik/icure/logic/impl/UserLogicImpl.java [hunk 2]
```diff
@@ -754,9 +754,9 @@ public User setProperties(User user, List<Property> properties) {
 			Optional<Property> prop = user.getProperties().stream().filter(pp -> pp.getType().getIdentifier().equals(p.getType().getIdentifier())).findAny();
 
 			if (!prop.isPresent()) {
-				user.getProperties().add(new Property(new PropertyType(p.getType().getType(), p.getType().getIdentifier()), new TypedValue(p.getType().getType(), p.getValue())));
```
Comment (aduchate) [inline src/main/java/org/taktik/icure/logic/impl/UserLogicImpl.java:757]: It seems to me the PropertyType type should come from the equivalent property in the passed parameter and not from the typed value

File: src/main/java/org/taktik/icure/logic/impl/UserLogicImpl.java (hunk 1 above)
Comment (aduchate) [reply src/main/java/org/taktik/icure/logic/impl/UserLogicImpl.java:759]: Then p.getTypedValue().getType() must only be used as a fallback when we look for the type for the property (not the typed value)

**Pull request merged**